### DISCUSSION TITLE
[Question] Calibration Woes

### DISCUSSION THREAD

**Discussion by mitch7391** [awawa-dev/HyperHDR#211] [Q&A]
Hi @awawa-dev, I do not have an exact issue but just some questions. I have just migrated over from Hyperion that I feel I have never been happy with and was so happy to find your solution that handles HDR. I instantly noticed a better handling of colours, but I still cannot get this project of mine in a state where I am not constantly adjusting it. I am wondering, in your setup, do you have this tuned to a point where you are no longer tweaking settings?

I have my Apple TV 4K to always force 4K HDR and have your project set to HDR turned on and my settings set how I like them. I then switch from Crunchyroll for example and over to Netflix or Disney+ on the same device and then the settings seem off; even though it is all forced as 4K HDR through the Apple TV (not set to match content). This confuses me and I would expect some differences between devices but not on the same device. I have even found there are times where last night we put on a colourful movie on Disney+ (Encanto) and the settings seemed perfect and then halfway through the movie it no longer seems right and colours are off (not quite matching in tone and at a few times like the gama settings were no longer right) and required me to tweak further during the movie before just giving up and turning off the lights. I have not changed anything to do with the LUT tables as it seriously confuses me and only adjusted through the video capturing hardware and the image processing settings.

I am not blaming the project and not upset about it (so I hope it does not read that way), as I never had the 'perfect' experience with Hyperion either and am just wondering if anyone actually has a perfect setup for all sources and if I should just give up on the Ambilight dream haha...

**Comment by awawa-dev** [comment]
Hi
Too many things could be affected including your hardware configuration. HDR tone mapping is based on bt2020 and hybrid gamma correction that are missing in the processing of almost all USB grabbers. Proven in action and you have included samples . Some grabber like Ezcap320 or AV Access do some partial brightness correction using unknown algorithms and they required special included LUT tables. The default one can't be use for them.

**Reply by mitch7391** [reply]
Hey @awawa-dev, thanks for the reply :) I figured there would be a lot of aspects that could affect my issue. So if I am understanding what you said correctly, I need to move away from a USB capture device and to something like the ezcap 320?

I guess what I was asking above in my ramble, do you have a perfect or near perfect setup? Where no matter what you watch is mapped accurately and does not require constant tweaking?

**Reply by awawa-dev** [reply]
No, MS2109 is just fine as it doesn't alternate the video stream in some own way. It can't just process meta-data which causes incorrect color (bt2020) and dim image (hybrid gamma). And that can be fixed with default LUT table. Speed is also good for USB2.0 device. As you can read it in my the grand latency test on my blog, ezcap 320 using USB3.0  is just ~10-20ms faster.

**Reply by awawa-dev** [reply]
Beside MS2109 also 4k60 HDR HDMI splitter would be needed. For example capable of removing HDCP &18gbps like VSP01201. ezcoo EZ-SP12H2 also works but 18gbps is not confirmed. But still the whole setup is cheaper and more capable than ezcap 320, because every grabber with a loop internal splitter (like this in ezcap 320 although it is high-end) is limited.

**Reply by mitch7391** [reply]
I have an ezcoo EZ-SP12HAS (the one with CEC) going into a OTHA HDMI to USB video capture card; plugged into a USB3 port on my RPi4 if that helps. I’m also using apa102 led strips.

**Reply by mitch7391** [reply]
Hey @awawa-dev I was just wondering if you could let me know how your system is in regards to my earlier question?

> do you have a perfect or near perfect setup? Where no matter what you watch is mapped accurately and does not require constant tweaking?

I am still trying to work out if the project is worth me pursuing any further or if what I am facing is expected limitations.

**Reply by awawa-dev** [reply]
I don't know such limitations. But there it such a lot of different custom setups: from the video player and used video format, through the video capturing system, to different LED outputs (mainly RGB vs RGBW) with different characteristic that I simply can't refer to each result even I would like to, because its subjective opinion.

**Reply by mitch7391** [reply]
Yep that makes sense and I understand that you can't speak for all of that. I am asking if you have achieved a perfect setup yourself? That your colours always match what is on the screen and you never have any issues with colours no longer matching across devices or SDR vs HDR content?

**Reply by awawa-dev** [reply]
I think I achieved it (but still see a room for new things), but again it's a personal opinion, this time from my side. I cannot imagine watching TV without what I found out when my equipment was damaged. And if I was dissatisfied with the result, I would probably never start this project instead of implementing more and more features.

**Reply by mitch7391** [reply]
Awesome, I think that is the encourage I needed to persist haha... Which capture device did you settle on in the end? I have flicked through the blog and it seems you have tested many.

**Reply by awawa-dev** [reply]
Currently I have Ezcap 320. Probably now I would go with MS2109 and FeinTech VSP01201 splitter, because it's cheaper and external HDMI splitter is always better solution. In latest v18beta there is a benchmark so I can test their performance and MS2109 is only just slightly slower than Ezcap. MS2109 is not capable of incoming 4k60 stream but FeinTech has a down-scaller.

**Reply by mitch7391** [reply]
Sweet, I will take a look at either of those, but as you said may start with the MS2109; I have bought countless products over the years (many of which faulty or fake) in trying to settle on the right product. Do you know if the FeinTech VSP01201 splitter is an better than the ezcoo EZ-SP12HAS that I already have? I do like that this one has CEC and we can continue to use the Apple TV remote for volume with it.

**Reply by awawa-dev** [reply]
Probably not much difference: both are similar and 18Gbps capable. FeinTech VSP01201 is many years on the market, has CEC support for first port and can be considered as stable solution. Ezcoo often provides software updates. But finally I don't know if its a advantage or disadvantage of Ezcoo because of the problematic firmware they sometimes release on the market (for example: don't even try to update ezcoo EZ-SP12H2 with latest flashing tool if you are not a hacker because of missing libraries on some systems).

**Reply by mitch7391** [reply]
> don't even try to update ezcoo EZ-SP12H2 with latest flashing tool if you are not a hacker because of missing libraries on some systems).

I did have this one and bricked it trying to update it, due to overseas delivery times during Covid, I was passed the return timeframe and they best they would do for me was a discount on buying another. I left this one as it was until I was sick of having sacrificed CEC for my Ambilight setup and eventually went to the more costly ezcoo EZ-SP12HAS; which was slightly less difficult to update and I did not brick it haha...

I will focus my attention on the grabbers you have recommended and look at LUT tables that suit them. Hopefully, I finally have a win :)
